### PR TITLE
fix(tx): support NEO3 transaction attributes

### PR DESCRIPTION
**Pull request by ixje**
The current code is still legacy Neo2. The transaction attributes have changed (and reduced) significantly. There are only 2 now
1. HighPriority
2. OracleResponse

* `High priority` can only be set by consensus committee members ([ref1](https://github.com/neo-project/neo/blob/77ee2cc5b6ea371efdf3be506b173c6304b0fc01/src/Neo/Network/P2P/Payloads/HighPriorityAttribute.cs#L38))
*  `OracleReponse` only by an entity voted into the network as OracleService by the consensus committee ([ref2](https://github.com/neo-project/neo/blob/77ee2cc5b6ea371efdf3be506b173c6304b0fc01/src/Neo/Network/P2P/Payloads/OracleResponse.cs#L103)). 
 
For these reasons I removed the ability to add attributes to a TX. Instead I only added support for (de)serialization because it seems liked the user [here](https://github.com/CityOfZion/neon-js/issues/876) was processing a TX with such attributes (likely an OracleResponse). Because I work so irregularly with JS/TS I took the style from `WitnessCondition`.

### CODE REVIEW

**Comment by melanke** [comment]
Task linked: [CU-2ve3xje update transaction attributes](https://app.clickup.com/t/2ve3xje)

File: packages/neon-core/src/tx/components/TransactionAttribute.ts [hunk 1]
```diff
@@ -1,139 +1,216 @@
...
-  // Base64-encoded data
-  data: string;
+export interface OracleResponseTransactionAttributeJson {
+  type: string;
+  // request id
+  id: number;
+  // response code
+  code: string;
+  // base64 encoded result for the request
+  result: string | HexString;
```
Comment (snowypowers) [inline packages/neon-core/src/tx/components/TransactionAttribute.ts:50]: if it is base64 encoded, it cannot be a HexString.

File: packages/neon-core/src/tx/components/TransactionAttribute.ts [hunk 2]
```diff
@@ -1,139 +1,216 @@
...
+  type: string;
 }
 
-export interface TransactionAttributeJson {
-  // Name of TransactionAttributeUsage in English
-  usage: string;
-  // Base64-encoded data
-  data: string;
+export interface OracleResponseTransactionAttributeJson {
+  type: string;
```
Comment (snowypowers) [inline packages/neon-core/src/tx/components/TransactionAttribute.ts:44]: in typescript, you can enforce this to be a static string "OracleResponse"  to be stricter. (Similarly for HighPriority)

Comment (ixje) [reply packages/neon-core/src/tx/components/TransactionAttribute.ts]: I think the change is what you asked for

File: packages/neon-core/src/tx/components/TransactionAttribute.ts [hunk 3]
```diff
@@ -1,139 +1,215 @@
...
-    if (TxAttrUsage.Url !== type) {
-      throw new Error(`${type} not found in TxAttrUsage!`);
-    }
-  }
+export type TransactionAttributeJson =
+  | HighPriorityTransactionAttributeJson
+  | OracleResponseTransactionAttributeJson;
 
-  return type as TxAttrUsage;
+export interface TransactionAttributeLike {
```
Comment (snowypowers) [inline packages/neon-core/src/tx/components/TransactionAttribute.ts:56]: is the `Like` interfaces being used? I cant seem to find them anywhere?

**Review by snowypowers — APPROVED** [review]

**Comment by ixje** [comment]
They’re used with the export() function. Which Transaction.export() calls on the attributes.

**Pull request merged**